src/libraries/Common/tests/TestUtilities/System/IO/PathGenerator.cs

```diff
@@ -17,6 +19,27 @@ public static string GenerateTestFileName([CallerMemberName] string memberName =
                 memberName ?? "TestBase",
                 lineNumber,
                 index.GetValueOrDefault(),
-                Guid.NewGuid().ToString("N").Substring(0, 8)); // randomness to avoid collisions between derived test classes using same base method concurrently
+                GenerateRandomFileSafeString(8)); // randomness to avoid collisions between derived test classes using same base method concurrently
+
+        private static string GenerateRandomFileSafeString(int length)
+        {
+            // A little more entropy than Guid.NewGuid().ToString("N").Substring(0, length))
+            const string alphanum = "abcdefghijklmnopqrstuvwxyz0123456789";
+
+            byte[] bytes = new byte[length];
+            lock (_rand)
```

Comment:
Just curious, what might happen if you don't `lock` it?

Reply:
The Random class is not thread safe - and this is a static method accessed from any thread running a test. (We do have Random.Shared which is thread safe, but it's not available when targeting .NET Framework)